src/com/gfredericks/schpec/numbers.clj

```diff
@@ -0,0 +1,119 @@
+(ns com.gfredericks.schpec.numbers
+  (:require [clojure.spec :as s]
+            [clojure.spec.gen :as gen])
+  (:import [java.math BigDecimal MathContext RoundingMode]))
+
+(def finite?
+  "Returns true if the given value is an actual finite number"
+  (let [falsies #{Double/POSITIVE_INFINITY
+                  Double/NEGATIVE_INFINITY
+                  Double/NaN}]
+    (fn [x]
+      (and (number? x)
+           (not (contains? falsies x))))))
+
+(s/def ::finite
+  (s/spec finite?
+          :gen #(gen/double* {:infinite? false :NaN? false})))
```

Comment:
ratios too probably

I realized after our conversations a couple weeks ago that a big reason I don't see a lot of utility in specs that combine exact and inexact types is that normal equality doesn't work; e.g., `(not= 42 42M)` and `(not= 42M 42.0)`. So I see the natural classes of numeric types as:

- fixed-integer < arbitrary-integer < ratio(including integers)
- bigdec
- doubles

where any of those five are natural classes to spec out, but combining those three categories is rather less useful.

So all that to say, I think restricting this to doubles makes a lot more sense.

Of course this gets a lot more muddled when we consider clojurescript :/